### ISSUE TITLE
Metadata_Renderer should use $parsed_post

### ISSUE THREAD

**Issue by nnlugovoy** [Parsely/wp-parsely#868]
## Describe the bug

in 3.3.0 - not using `$parsed_post`
```diff
diff --git a/src/UI/class-metadata-renderer.php b/src/UI/class-metadata-renderer.php
index 494304e..e896695 100644
--- a/src/UI/class-metadata-renderer.php
+++ b/src/UI/class-metadata-renderer.php
@@ -99,17 +99,17 @@ final class Metadata_Renderer {
                // We can't construct the metadata without a valid post object.
                $parsed_post = get_post( $post );
                if ( ! $parsed_post instanceof WP_Post ) {
                        return;
                }
 
                // Assign default values for LD+JSON
                // TODO: Mapping of an install's post types to Parse.ly post types (namely page/post).
-               $metadata = ( new Metadata( $this->parsely ) )->construct_metadata( $post );
+               $metadata = ( new Metadata( $this->parsely ) )->construct_metadata( $parsed_post );
 
                // Something went wrong - abort.
                if ( 0 === count( $metadata ) || ! isset( $metadata['headline'] ) ) {
                        return;
                }
 
                // Insert JSON-LD or repeated metas.
                if ( 'json_ld' === $meta_type ) {
```

in previous versions - using `$parsed_post` correctly

```diff
 $parsed_post = get_post( $post );
        if ( ! $parsed_post instanceof WP_Post ) {
            return;
        }

        // Assign default values for LD+JSON
        // TODO: Mapping of an install's post types to Parse.ly post types (namely page/post).
        $parsely_page = $this->construct_parsely_metadata( $parsely_options, $parsed_post );
```

**Comment by GaryJones** [comment]
Hi @nnlugovoy - thank you for reporting this. 

Good catch on this change! 

Have you come across any situations where the global `$post` hasn't already been an instance of `WP_Post`, and has therefore created a fatal error when passed into the [`method`](https://github.com/Parsely/wp-parsely/blob/b780f51ba944d92be08ead9a705af23c7d18b80c/src/class-metadata.php#L46)?

**Comment by nnlugovoy** [comment]
I tracked it down to a very simple reproducible sequence:

```php
global $post; 
query_posts(["fields" => "ids"]);
while(have_posts()) the_post();
rewind_posts();
var_dump($post);

int(2048867)
```
The catch is `'fields' => 'ids'`

A real use case that revealed this bug:
1. page used a custom query to fetch some relevant post ids
1. and custom plugin was checking posts from the current query to determine additional JS  in `wp_enqueue_scripts` action
2. then `enqueue_js_tracker` from `wp-parsely`  was called for the same action, having int value `$post = 2048867;` and crashing

**Comment by GaryJones** [comment]
@nnlugovoy Thank you for the use case. 

I'd hope that after custom queries, the [`wp_reset_postdata()`](https://developer.wordpress.org/reference/functions/wp_reset_postdata/) function would be called, but know that's not always going to happen. That in turn calls [this method](https://developer.wordpress.org/reference/classes/wp_query/reset_postdata/), which resets the global `$post`.

Easy enough to accommodate in the plugin though, by fixing the bug you identified. 👍🏻

**Comment by nnlugovoy** [comment]
Maybe it is a bit more complicated, as `wp_reset_postdata()` still puts the first ID to the `$post` :

`wp eval 'global $post; query_posts(["fields" => "ids"]); while(have_posts()) the_post(); rewind_posts(); wp_reset_postdata(); do_action("wp_head");'`

```c
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Parsely\Metadata::construct_metadata() must be an instance of WP_Post, int given, called in /var/www/html/wp-content/plugins/wp-parsely-3.3/src/UI/class-metadata-renderer.php on line 108 and defined in /var/www/html/wp-content/plugins/wp-parsely-3.3/src/class-metadata.php:46
Stack trace:
#0 /var/www/html/wp-content/plugins/wp-parsely-3.3/src/UI/class-metadata-renderer.php(108): Parsely\Metadata->construct_metadata(2048867)
#1 /var/www/html/wp-content/plugins/wp-parsely-3.3/src/UI/class-metadata-renderer.php(68): Parsely\UI\Metadata_Renderer->render_metadata('json_ld')
#2 /var/www/html/wp-includes/class-wp-hook.php(303): Parsely\UI\Metadata_Renderer->render_metadata_on_head('')
#3 /var/www/html/wp-includes/class-wp-hook.php(327): WP_Hook->apply_filters(NULL, Array)
#4 /var/www/html/wp-includes/plugin.php(470): WP_Hook->do_action(Array)
#5 /root/.composer/vendor/wp-cli/eval-command/src/Eval_Command.php(39) : eval()'d code(1): do_action('wp_head')
#6 /root/.composer/vendo in /var/www/html/wp-content/plugins/wp-parsely-3.3/src/class-metadata.php on line 46
```

**Comment by acicovic** [comment]
Hi @nnlugovoy, and thank you for reporting this.

This is now fixed in #869, and we should have a fixed release (3.3.2) in the beginning of next week.

Thank you!
